sdks/go/pkg/beam/core/runtime/exec/optimized/gen.go

```diff
@@ -0,0 +1,27 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package optimized
+
+// NOTE(herohde) 1/18/2018: omit []byte for encodes to avoid re-registration due to symmetry.
```

Comment:
<!--new_thread; commit:9c2390201ba3ecb080c84c7963396368ee720ae0; resolved:0-->
Is this a TODO, or is it something that's actually done already? It sounds like a TODO.

Reply:
It's already done. The command for encoders use "primitives" instead of "data".